cdap-master/src/main/java/co/cask/cdap/data/runtime/main/MasterServiceMain.java

```diff
@@ -205,6 +205,8 @@ public MasterServiceMain() {
     login(cConf);
 
     Configuration hConf = HBaseConfiguration.create();
+    hConf.set(Constants.HBase.CLIENT_RETRIES, cConf.get(Constants.HBase.MASTER_CLIENT_RETRIES));
```

Comment:
The reason of not setting properties programmatically in the TwillRunnable is because it's hard to tell what properties get changed. If we set everything from the master side, then we can trust the settings in the localized `hConf.xml` and `cConf.xml` files in the container directory, making debugging much easier.